contrib/lncli.fish

```diff
@@ -0,0 +1,40 @@
+# fish programmable completion for lncli
+# copy to $HOME/.config/fish/completions and restart your shell session
+
+# TODO: provide available arguments for subcommands
+
+complete -c lncli -d "print the version" -l version -s v
+complete -c lncli -d "show help" -l help -s h
+complete -c lncli -d "if set, lock macaroon to specific IP" -l macaroonip -xf
+complete -c lncli -d "anti-replay macaroon validity time in seconds (default: 60)" -l macaroontimeout -xf
+complete -c lncli -d "path to macaroon file" -l macaroonpath -x
+complete -c lncli -d "disable macaroon authentication" -l no-macaroons
+complete -c lncli -d 'the network lnd is running on' -l network -s n -x -a "mainnet testnet regtest simnet"
+complete -c lncli -d 'the chain lnd is running on' -l chain -s c -x -a "bitcoin litecoin"
+complete -c lncli -d 'path to TLS certificate' -l tlscertpath -x
+complete -c lncli -d "path to lnd's base directory" -l lnddir -x
+complete -c lncli -d "host:port of ln daemon" -l rpcserver -xf
+
+# get the list of all available subcommands
+# in the help section, all subcommands start 
+# with 4 spaces
+set commands (lncli help | grep '^    ' | grep -v help | sed -e 's/^[[:space:]]*//' | cut -f 1 -d " ")
```

Comment:
Maybe we could use the same `awk` command here that is already used for bash completion?
`lncli help 2>/dev/null | awk '$1 ~ /^[a-z]/ { print $1; }'`
That might be a bit more resilient against changes in how the help screen is formatted.